src/views/deckviewer/behavior.tsx

```diff
@@ -296,24 +304,39 @@ class DeckViewerViewBehavior extends ViewBehavior<DeckViewerIncludedData> {
       this.saveDeckChange();
     } else if (action === "similar") {
       console.log("looking for similar cards");
-      (document.querySelector(
-        "#filterSelection > div > ul > li[data-active=true]"
-      ) as HTMLElement | null)?.click();
-      (document.querySelector(
-        "#filterSelection > div > ul > li[data-filtertype=misc]"
-      ) as HTMLElement | null)?.click();
-
-      this.cardSearchUtil!.GetNameFilter().setValue(
-        this.dl.getMapData("IDToName")![cardId]
-      );
-      this.cardSearchUtil!.GetMiscFilter().setValue(["Show Duplicates"]);
-      this.cardSearchUtil!.ApplyFilter();
+      this.altPane.open(this.dl.getMapData("IDToName")![cardId]);
     } else if (action === "tosideboard") {
       this.onSideboardAction("add", cardId);
       this.onSearchAction("remove", cardId);
       this.saveDeckChange();
     } else if (action === "star") {
       this.updateKeyCard(cardId);
+    } else if (action === "replaceall") {
+      const replaceables = this.dl.getMapData("NameToID")![
+        this.dl.getMapData("IDToName")![cardId]
+      ];
+      const toBeReplaced: Record<string, number> = {};
+      let prevCard = "";
+      for (const card of deckDetails.mainboard) {
+        if (replaceables.indexOf(card) > -1 && card !== cardId) {
+          if (prevCard === card) {
+            toBeReplaced[card]++;
+          } else {
+            toBeReplaced[card] = 1;
+          }
+          prevCard = card;
+        }
+      }
+      for (const cardName in toBeReplaced) {
+        for (let i = 0; i < toBeReplaced[cardName]; i++) {
+          this.onSearchAction("remove", cardName);
```

Comment:
We shouldn't call `this.onSearchAction("remove", ...)` in a loop. Each removal updates the mainboard renderer, which is gonna be slow on big decks.

Also, just for reference, "onSearchAction" operates only on the mainboard. When you add interactions with the sideboard you'll need "onSideboardAction".